agent/php_error.c

```diff
@@ -423,12 +423,15 @@ static int nr_php_should_record_error(int type, const char* format TSRMLS_DC) {
    * conditional will help us prevent double capture once we start recording
    * more than one error per transaction.
    *
-   * Note:  The format string comparison is a fragile check:  This "Uncaught %s"
+   * Note:  The format string comparison is a fragile check:  This "Uncaught"
    * string is not guaranteed in the PHP runtime.  If there was a better way to
    * detect uncaught exceptions we would do so.  To reduce the chance that this
    * early exit triggers erroneously, we check for the exception error type.
+   *
+   * In PHP 8.0+, this is no longer a format string so we now omit the %s in the
+   * check.
    */
-  if ((E_ERROR == type) && (0 == nr_strnicmp(format, NR_PSTR("Uncaught %s")))) {
```

Comment:
Does this need a condition for < PHP8? Or is the backward compatibility not an issue here?

Reply:
Backwards compatibility isn't an issue here. In PHP < 8.0, this is a format string ("Uncaught %s thrown") while in PHP 8.0+, the format string is expanded by the time we get here ("Uncaught <the full exception stacktrace> thrown"). This change just shortens the string we look for so it is the string the common across all PHP versions.